tests/cephfs/snapshot_clone/snap_schedule_with_mds_restart.py

```diff
@@ -0,0 +1,273 @@
+        if not fs_details:
+            fs_util.create_fs(client1, default_fs)
+        subvolumegroup_list = [
+            {"vol_name": default_fs, "group_name": "subvolgroup_snap_schedule"},
+        ]
+        for subvolumegroup in subvolumegroup_list:
+            fs_util.create_subvolumegroup(client1, **subvolumegroup)
+        log.info("Kernel mount")
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util.get_mon_node_ips()
+        retry_mount = retry(CommandFailed, tries=3, delay=30)(fs_util.kernel_mount)
+        retry_mount(
+            [client1],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            extra_params=f",fs={default_fs}",
+        )
+        log.info("Run IO")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 400 "
+            f"--files 100 --files-per-dir 10 --dirs-per-dir 2 --top "
+            f"{kernel_mounting_dir_1}",
+            long_running=True,
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"mkdir -p {kernel_mounting_dir_1}/dir_kernel"
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"mkdir -p {kernel_mounting_dir_1}/snap_schedule"
+        )
+        log.info("Fuse mount")
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse{mounting_dir}_1/"
+        fs_util.fuse_mount(
+            [client1],
+            fuse_mounting_dir_1,
+            extra_params=f"--client_fs {default_fs}",
+        )
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_1}/dir_fuse")
+        sanp_schedule_list = ["/dir_kernel", "/dir_fuse"]
+        m_granularity = (
+            "m" if LooseVersion(ceph_version) >= LooseVersion("17.2.6") else "M"
+        )
+        log.info("Verify Ceph Status is healthy before starting test")
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=300)
+        ceph_healthy = 0
+        while (datetime.datetime.now() < end_time) and (ceph_healthy == 0):
+            try:
+                fs_util.get_ceph_health_status(client1)
+                ceph_healthy = 1
+            except Exception as ex:
+                log.info(ex)
+                log.info("Wait for few secs and recheck ceph status")
+                time.sleep(5)
+        if ceph_healthy == 0:
+            assert False, "Ceph remains unhealthy even after wait for 300secs"
+        commands = [
+            f"ceph fs subvolume ls {default_fs}",
+            "ceph config set mgr mgr/snap_schedule/allow_m_granularity true",
+            f"ceph fs snap-schedule add path 1{m_granularity} --fs {default_fs}",
+        ]
+        modified_commands = [
+            cmd.replace("path", item) for item in sanp_schedule_list for cmd in commands
+        ]
+
+        for cmd in modified_commands:
+            client1.exec_command(sudo=True, cmd=cmd)
+        sleep(300)
```

Comment:
Sleeping for 5 minutes...

Reply:
Hi Satya,
We've set snapshots to be created every 1 minute, so in the `verify_snap_schedule` method, we check if at least 4 snapshots are there. To make sure they have time to be created, we wait for 5 minutes.